File: test.spritebuilder/Source/TDEngine/Engine/MoneyEngine.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package Engine;

/**
 *
 * @author Константин
 */
class MoneyEngine {

    public MoneyEngine(int money) {
        this.money = money;
    }
int money;

    public boolean canPay(int cost)
    {
        return money > cost;
    }
    
    public void Pay(int cost)
    {
    money -= cost;
    }
    
    public void fillMoney(int count)
    {
      money += count;
    }

    int getMoney() {
       return money;
    }

 
    
    
}

File: TowerDefence.spritebuilder/Engine/MoneyEngine.swift
//
//  MoneyEngine.swift
//  TowerDefence
//
//  Created by koveres on 07.12.14.
//  Copyright (c) 2014 Apportable. All rights reserved.
//

import Foundation
public class MoneyEngine {
    
    public init( money_:CInt) {
         money = money_;
    }
    var money:CInt;
    
    public  func canPay(  cost:CInt) ->(Bool)
    {
        return (money > cost);
    }
    
    public func Pay(  cost:CInt)
    {
        money -= cost;
    }
    
    public func fillMoney(  count:CInt)
    {
        money += count;
    }
    
    func getMoney() -> (CInt){
        return money;
    }  
}
